src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioWindowsSolutionStatusService.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.OperationProgress;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(RazorSolutionStatusService))]
+    internal class VisualStudioWindowsSolutionStatusService : RazorSolutionStatusService
+    {
+        private readonly IVsOperationProgressStatusService? _operationProgressStatusService;
+
+        [ImportingConstructor]
+        public VisualStudioWindowsSolutionStatusService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider!!)
+        {
+            if (serviceProvider.GetService(typeof(SVsOperationProgress)) is IVsOperationProgressStatusService service)
+            {
+                _operationProgressStatusService = service;
+            }
+        }
+
+        public override bool TryGetIntelliSenseStatus([NotNullWhen(returnValue: true)] out RazorSolutionStatus? status)
+        {
+            if (_operationProgressStatusService == null)
```

Comment:
Either this check can be removed or the `?.` below can be removed.

Reply:
Great point! I'll remove the `?.` below. I prefer the explicit opt-outs anyhow. Makes it easier to debug 😄